HCL.IdentityServer.API.Test/IntegrationTest/Services/RegistrationServiceIntegrationTest.cs

```diff
@@ -0,0 +1,141 @@
+
+            var accountForRegistration = new AccountDTO()
+            {
+                Login = "Ilia1",
+                Password = "123456",
+            };
+
+            //Act
+            var authDTO = await regServ.Registration(accountForRegistration);
+
+            //Assert
+            authDTO.Should().NotBeNull();
+            authDTO.Data.Should().NotBeNull();
+            authDTO.StatusCode.Should().Be(StatusCode.AccountCreate);
+        }
+
+        [Fact]
+        public async Task Registration_WithAlredyExistAccount_ReturnExistStatusCode()
+        {
+            //Arrange
+            using var scope = webHost.Services.CreateScope();
+            var appDBContext = scope.ServiceProvider.GetRequiredService<AppDBContext>();
+            var accRep = new AccountRepository(appDBContext);
+            var accountServ = new AccountService(accRep, StandartMockBuilder.mockLoggerAccServ);
+            var tokServ = new TokenService(StandartMockBuilder.jwtOpt);
+            var regServ = new RegistrationService(accountServ, tokServ, StandartMockBuilder.mockLoggerRegServ);
+
+            var accountForRegistration = new AccountDTO()
+            {
+                Login = "Ilia2",
+                Password = "123456",
+            };
+
+            //Act
+            await regServ.Registration(accountForRegistration);
+            var authDTO = await regServ.Registration(accountForRegistration);
+
+            //Assert
+            authDTO.Should().NotBeNull();
+            authDTO.StatusCode.Should().Be(StatusCode.AccountExist);
+        }
+
+
+        [Fact]
+        public async Task AuntificationAccount_WithRightAuthData_ReturnAuthenticateStatusCode()
+        {
+            //Arrange
+            using var scope = webHost.Services.CreateScope();
+            var appDBContext = scope.ServiceProvider.GetRequiredService<AppDBContext>();
+            var accRep = new AccountRepository(appDBContext);
+            var accountServ = new AccountService(accRep, StandartMockBuilder.mockLoggerAccServ);
+            var tokServ = new TokenService(StandartMockBuilder.jwtOpt);
+            var regServ = new RegistrationService(accountServ, tokServ, StandartMockBuilder.mockLoggerRegServ);
+
+            var accountForRegistration = new AccountDTO()
+            {
+                Login = "Ilia3",
+                Password = "123456",
+            };
+
+            //Act
+            await regServ.Registration(accountForRegistration);
+            var authDTO = await regServ.Authenticate(accountForRegistration);
+
+            //Assert
+            authDTO.Should().NotBeNull();
+            authDTO.StatusCode.Should().Be(StatusCode.AccountAuthenticate);
+        }
+
+        [Fact]
+        public async Task AuntificationAccount_WithWrongAuthData_ReturnKeyNotFoundException()
+        {
+            //Arrange
+            using var scope = webHost.Services.CreateScope();
```

Comment:
You can initialize this variables in constructor and then use it or make separate method with getting services, repos. You have repeated block of code